cinema/views.py

```diff
@@ -51,8 +83,58 @@ class MovieSessionViewSet(viewsets.ModelViewSet):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieSessionListSerializer
-
         if self.action == "retrieve":
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("cinema_hall")
+                .annotate(tickets_available=F(
+                    "cinema_hall__rows"
+                ) * F(
+                    "cinema_hall__seats_in_row"
+                ) - Count("tickets"))
+            )
+
+        if date:
+            date_list = date.split("-")
+            date = datetime.date(
+                int(date_list[0]),
+                int(date_list[1]),
+                int(date_list[2])
+            )
+            queryset = queryset.filter(show_time__startswith=date)
```

Comment:
You can just use show_time__date

Reply:
The checklist was not read!
https://github.com/AlexandrZhezheria/py-tickets-orders/blob/develop/checklist.md